agent/pool/pool.go

```diff
@@ -319,6 +320,11 @@ func DialTimeoutWithRPCType(dc string, addr net.Addr, src *net.TCPAddr, timeout
 			return nil, nil, err
 		}
 		conn = tlsConn
+
+		// Cast to tls.Conn to expose CloseWrite method on HalfCloser
```

Comment:
```suggestion
		// If this is a tls.Conn, expose HalfCloser to caller
```
Super nit picky but it's not very idiomatic to talk about "casting" in Go